src/libasr/pass/pass_list_concat.cpp

```diff
@@ -0,0 +1,70 @@
+#include <libasr/asr.h>
+#include <libasr/containers.h>
+#include <libasr/exception.h>
+#include <libasr/asr_utils.h>
+#include <libasr/asr_verify.h>
+#include <libasr/pass/pass_list_concat.h>
+#include <libasr/pass/pass_utils.h>
+
+#include <vector>
+#include <utility>
+
+
+namespace LFortran {
+
+using ASR::down_cast;
+using ASR::is_a;
+
+/*
+This ASR pass replaces list-concat to
+
+Converts:
+    x = i + j // lists
+
+to:
+
+ */
+
+
+class ReplaceListConcat : public ASR::BaseExprReplacer<ReplaceListConcat>
+{
+    private:
+
+        Allocator& al;
+
+    public:
+        ReplaceListConcat(Allocator &al_) : al(al_)
+        { }
+
+        void replace_ListConcat(const ASR::ListConcat_t* x) {
+            Location loc = x->base.base.loc;
+            *current_expr = nullptr;
+        }
+
+};
+
+class ListConcatVisitor : public ASR::CallReplacerOnExpressionsVisitor<ListConcatVisitor>
+{
+    private:
+
+        ReplaceListConcat replacer;
+
+    public:
+
+        ListConcatVisitor(Allocator& al_) : replacer(al_) {}
+
+        void call_replacer() {
+            replacer.current_expr = current_expr;
+            replacer.replace_expr(*current_expr);
+        }
+
+};
+
+void pass_list_concat(Allocator &al, ASR::TranslationUnit_t &unit) {
```

Comment:
You are not accepting PassOptions. See how other passes are declared and defined. Follow the same pattern here.